src/sinks/aws_s3/config.rs

```diff
@@ -176,13 +178,24 @@ impl S3SinkConfig {
 
         // Configure our partitioning/batching.
         let batch_settings = self.batch.into_batcher_settings()?;
-        let key_prefix = self
+        let key_prefix: Template = self
             .key_prefix
             .as_ref()
             .cloned()
             .unwrap_or_else(|| DEFAULT_KEY_PREFIX.into())
             .try_into()?;
-        let partitioner = KeyPartitioner::new(key_prefix);
+        let ssekms_key_id: Option<Template> = if self.options.ssekms_key_id.is_none() {
+            None
+        } else {
+            self.options
+                .ssekms_key_id
+                .as_ref()
+                .cloned()
+                .map(Template::try_from)
+                .unwrap()
+                .ok()
+        };
```

Comment:
@StephenWakely I'm trying to essentially convert an `Option<String>` into an `Option<Template>` here, leavinig a value of `None` as-is...ran into a lot of issues with unwrapping when a `None` is involved so I went with this conditional here.  I'm feeling like there's probably a more elegant way to do this in Rust, but I'm a bit of a novice so if you know of a better way to organize this kind of thing let me know.